Tools/CMake/AMReXGenexHelpers.cmake

```diff
@@ -279,7 +279,7 @@ function ( eval_genex _list _lang _comp )
       string(REGEX REPLACE "\\$<PLATFORM_ID:[A-Za-z]*>" "0"  _in "${_in}")
 
       # Genex in the form $<*_COMPILER_ID:compiler_ids>
-      string(REGEX REPLACE "\\$<${_lang}_COMPILER_ID:[^>]*${_comp}[^>]*>" "1"  _in "${_in}")
+      string(REGEX REPLACE "\\$<${_lang}_COMPILER_ID[^>]*[:,]${_comp}[>,]" "1"  _in "${_in}")
```

Comment:
Can you help me understand what the change in regex does for each part?
- Does `[:,]` now match a literal `:` or `,`?
- Does `[>,]` now match a literal `>` or `,`?

Reply:
> * Does `[:,]` now match a literal `:` or `,`?
> * Does `[>,]` now match a literal `>` or `,`?

Yes to both, the idea is that we will now exactly (without prefixes or suffixes) match `${_comp}` because it must be surrounded by commas (if it's in the middle of the list) or immediately follow the colon at the beginning of the list or immediately precede the chevron at the end of the list.

Reply:
It would also match `$<CXX_COMPILER_ID,someid>`, right?

Reply:
Thanks! So why do we need to match the `,`?